store/tikv/client_helper.go

```diff
@@ -73,13 +74,22 @@ func (ch *ClientHelper) ResolveLocks(bo *Backoffer, callerStartTS uint64, locks
 }
 
 // SendReqCtx wraps the SendReqCtx function and use the resolved lock result in the kvrpcpb.Context.
-func (ch *ClientHelper) SendReqCtx(bo *Backoffer, req *tikvrpc.Request, regionID RegionVerID, timeout time.Duration, sType kv.StoreType, directStoreAddr string) (*tikvrpc.Response, *RPCContext, string, error) {
+func (ch *ClientHelper) SendReqCtx(bo *Backoffer, req *tikvrpc.Request, regionID RegionVerID, timeout time.Duration, sType kv.StoreType, directStoreAddr string, opts ...StoreSelectorOption) (*tikvrpc.Response, *RPCContext, string, error) {
 	sender := NewRegionRequestSender(ch.regionCache, ch.client)
 	if len(directStoreAddr) > 0 {
 		sender.SetStoreAddr(directStoreAddr)
 	}
 	sender.Stats = ch.Stats
 	req.Context.ResolvedLocks = ch.resolvedLocks.GetAll()
-	resp, ctx, err := sender.SendReqCtx(bo, req, regionID, timeout, sType)
+	failpoint.Inject("assertStaleReadFlag", func(val failpoint.Value) {
+		if val.(bool) {
+			if len(opts) > 0 {
+				if !req.StaleRead {
+					panic("req.StaleRead should be true")
+				}
+			}
```

Comment:
How about the situation that `opts` must be empty when `StaleRead` is false?

Reply:
I think this assertion is equal to your situation.

Reply:
Your case is `len(opts) > 0 && !req.StaleRead`, my case is `len(opts) == 0 && req.StaleRead`.

Reply:
I think `len(opts) == 0 && req.StaleRead` is valid, it indicates a staleness transaction with global txnScope.